primitives/api/proc-macro/src/impl_runtime_apis.rs

```diff
@@ -139,14 +170,28 @@ fn generate_impl_calls(
 
 		for item in &impl_.items {
 			if let ImplItem::Fn(method) = item {
-				let impl_call =
-					generate_impl_call(&method.sig, &impl_.self_ty, input, &impl_trait)?;
+				let impl_call = generate_impl_call(
+					&method.sig,
+					&impl_.self_ty,
+					input,
+					&impl_trait,
+					&trait_api_ver,
+				)?;
+				let mut attrs = filter_cfg_attrs(&impl_.attrs);
+
+				let method_api_version = extract_api_version(&method.attrs, method.span())?;
+				if method_api_version.custom.is_some() {
+					return Err(Error::new(method.span(), "`api_version` attribute is not allowed `decl_runtime_api` method implementations"))
+				}
+				if let Some(feature_gated) = method_api_version.feature_gated {
+					add_feature_guard(&mut attrs, &feature_gated.0, true)
+				}
```

Comment:
Isn't this already checked in the `ApiImplItem` code?

Reply:
You mean why I don't use `ApiImplItem` to add the feature flags? `impl_call` is `TokenStream` here. `ApiImplItem` works with `ItemImpl`.